editor/src/core/shared/uid-utils.ts

```diff
@@ -208,3 +212,33 @@ export function fixUtopiaElement(
 
   return fixUtopiaElementInner(elementToFix)
 }
+
+export function uidsFromString(uidList: string = ''): Array<string> {
+  return uidList.split(' ')
+}
+
+export function uidsToString(uidList: Array<string>): string {
+  return uidList.join(' ')
+}
+
+export function appendToUidString(
+  uidsString: string | null | undefined,
+  uidsToAppend: string | null | undefined,
+): string | null {
+  if (uidsToAppend == null) {
+    return uidsString ?? null
+  } else if (uidsString == null || uidsString.length === 0) {
+    return uidsToAppend
+  } else if (R.difference(uidsFromString(uidsToAppend), uidsFromString(uidsString)).length > 0) {
```

Comment:
This is probably a very expensive function. The docs mention using "value equality, not reference equality", implying it's probably using the extra expensive `R.equals` function.

Reply:
I'll take this